### PR TITLE
Refactor [v114] Renew telemetry

### PR DESCRIPTION
**Pull request by dnarcese**
# Request for Data Collection Renewal

** This form is for the renewal of an existing, reviewed data collection.**

** All questions are mandatory.
You must receive Data Review from a
[Data Steward](https://wiki.mozilla.org/Data_Collection)
on a filled-out Request before shipping your renewed data collection.**

1) Provide a link to the initial Data Collection Review Request for this collection.
https://github.com/mozilla-mobile/firefox-ios/pull/12334
2) When will this collection now expire?
2023-12-01
3) Why was the initial period of collection insufficient?
Still need the data

### CODE REVIEW

File: Client/metrics.yaml [hunk 1]
```diff
@@ -161,35 +170,10 @@ app_menu:
...
-        Counts the number of times a user taps Home in the app menu
-    bugs:
-      - https://github.com/mozilla-mobile/firefox-ios/issues/8147
-    data_reviews:
-      - https://github.com/mozilla-mobile/firefox-ios/pull/8167
-      - https://github.com/mozilla-mobile/firefox-ios/pull/9673
-      - https://github.com/mozilla-mobile/firefox-ios/pull/12334
-    notification_emails:
-      - fx-ios-data-stewards@mozilla.com
-    expires: "2023-06-01"
```
Comment (lmarceau) [inline Client/metrics.yaml:192]: AppMenu.library and AppMenu.home shouldn't be removed no? 🤔

Comment (dnarcese) [reply Client/metrics.yaml:192]: Unless I'm missing something, they don't exist anymore?
Or are we still using the same Home button probe even though it is in the toolbar now?

Comment (lmarceau) [reply Client/metrics.yaml:192]: Home is [still used](https://github.com/mozilla-mobile/firefox-ios/blob/3db84a24be39af6d453c02574addfcc45a6f4526/Client/Frontend/Browser/BrowserViewController/BrowserViewController%2BTabToolbarDelegate.swift#L17) in `tabToolbarDidPressHome` function of `BrowserViewController`. 

As for library it seems we can remove it, but we should clean up `TelemetryWrapper`
Screenshot 2023-04-24 at 2 34 57 PM

**Comment by lmarceau** [comment]
The changes LGTM, but the build failed due to glean parser errors:
```
    Sorry, Glean found some glinter nits:
    /Users/vagrant/git/Client/metrics.yaml (177:4) - syntax error: expected <block end>, but found '<block mapping start>' (syntax)
    /Users/vagrant/git/Client/metrics.yaml (178:5) - wrong indentation: expected 5 but found 4 (indentation)
    Please fix the above nits to continue.
    ==============================================================================
    /Users/vagrant/git/Client/metrics.yaml:
        while parsing a block mapping   in
        "/Users/vagrant/git/Client/metrics.yaml", line 151, column 3 expected
        <block end>, but found '<block mapping start>'   in
        "/Users/vagrant/git/Client/metrics.yaml", line 177, column 4
    ERROR running glean_parser v7.1.0
    ❌ Found 3 errors.
❌  error: glean_parser failed. See errors above.
```

**Review by lmarceau — APPROVED** [review]

**Pull request merged**